spec/integration/non_retryable_exception/non_retryable_exception_spec.rb

```
@@ -0,0 +1,146 @@
+  def perform(sqs_msg, body)
+    receive_count = sqs_msg.attributes['ApproximateReceiveCount'].to_i
+    DT[:attempts] << { receive_count: receive_count, body: body }
+
+    case body
+    when 'non_retryable_invalid'
+      raise InvalidInputError, 'Invalid input data'
+    when 'non_retryable_not_found'
+      raise RecordNotFoundError, 'Record not found'
+    when 'retryable_error'
+      # Fail on first attempt, succeed on retry
+      if receive_count < 2
```

RuboCop (1.84.0)

[convention] Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.

(Style/IfUnlessModifier)